src/core/api/serializers.py

```diff
@@ -19,3 +20,40 @@ class FeedbackSerializer(serializers.ModelSerializer):
     class Meta:
         model = Feedback
         fields = '__all__'
+
+
+class FeedbackListSerializer(serializers.ModelSerializer):
+    course_code = serializers.CharField(source='assignment.course.code')
+    course_name = serializers.CharField(source='assignment.course.name')
+    ta_supervisor = serializers.CharField(source='assignment.course.supervisor')
+    status = serializers.SerializerMethodField()
+    due_by = serializers.ReadOnlyField(default="--")
+
+    def get_status(self, value):
```

Comment:
**Tip:** Maybe just do an `option` verb first to get the details of status' key and value. If this is a read-only value for frontend then this is absolutely fine.